src/main/java/edu/tamu/app/service/manager/GitHubProjectService.java

```diff
@@ -26,46 +24,43 @@ public GitHubProjectService(final ManagementService managementService) throws IO
     public List<Sprint> getActiveSprintsByScopeId(final String scopeId) throws Exception {
         logger.info("Fetching active sprints for remote project with scope id " + scopeId);
         GHRepository repo = github.getRepositoryById(scopeId);
-        String projectName = repo.getName();
-        return repo.listProjects(ProjectStateFilter.OPEN)
-            .asList().stream()
-            .filter(p -> p.getName().toUpperCase().contains(SPRINT))
-            .map(p -> new Sprint(String.valueOf(p.getId()), p.getName(), projectName, getCards(p)))
+        String productName = repo.getName();
+        return repo.listProjects(ProjectStateFilter.OPEN).asList().stream()
+            .filter(this::isSprintProject)
+            .map(p -> toSprint(p, productName))
             .collect(Collectors.toList());
     }
 
     @Override
     public List<Sprint> getAdditionalActiveSprints() throws Exception {
-        GHOrganization organization = github.getOrganization(ORGANIZATION);
-        return organization.listProjects(ProjectStateFilter.OPEN).asList().stream()
-            .filter(p -> p.getName().toUpperCase().contains(SPRINT))
-            .map(p -> new Sprint(
-                String.valueOf(p.getId()),
-                String.format("%s - %s", ORGANIZATION, p.getName()),
-                ORGANIZATION,
-                getCards(p)
-            ))
+        return github.getOrganization(ORGANIZATION)
+            .listProjects(ProjectStateFilter.OPEN).asList().stream()
+            .filter(this::isSprintProject)
+            .map(p -> toSprint(p, ORGANIZATION))
             .collect(Collectors.toList());
     }
 
+    private boolean isSprintProject(GHProject project) {
+        return project.getName().toUpperCase().contains(SPRINT);
+    }
+
+    private Sprint toSprint(GHProject project, String productName) {
+        return new Sprint(
+            String.valueOf(project.getId()),
+            toProductName(project),
+            productName,
+            getCards(project)
+        );
+    }
+
     private List<Card> getCards(GHProject project) {
         return exceptionHandlerWrapper(project, i -> i.listColumns().asList().stream())
             .flatMap(c -> exceptionHandlerWrapper(c, i -> i.listCards().asList().stream()))
-            .collect(HashMap<GHProjectCard, GHIssue>::new, (m, c) -> m.put(c, exceptionHandlerWrapper(c, i -> i.getContent())), HashMap<GHProjectCard, GHIssue>::putAll)
```

Comment:
No reason to collect and then stream again.

Reply:
This collect to a map with key GHProjectCard was causing loss of a sprint card due to key overwrite of a duplicate GHProjectCard.

```
-private static final GHProjectCard TEST_CARD1 = mock(GHProjectCard.class);
+private static final GHProjectCard TEST_CARD1 = mock(GHProjectCard.class, RETURNS_DEEP_STUBS.get());
```